File: test/fixtures/vuln-api.js
const { ApolloServer } = require("apollo-server");
const WebSocket = require("ws");
const axios = require("axios");
const _ = require("lodash");

const server = new ApolloServer({ introspection: true, depthLimit: 0, complexity: 0 });
const wss = new WebSocket.Server({ verifyClient: (info, cb) => cb(true) });

_.merge(config, req.body);
obj["__proto__"] = {};
User.find({ $where: "this.name == '" + req.body.name + "'" });
User.find({ age: { $gt: req.body.age } });
ldap.search("cn=" + req.query.name);
res.render(req.body.template);
axios.get(req.body.url);
fetch(req.body.url);

console.log("debug");
const token = "hardcoded123";
res.redirect("http://api.internal.svc:8080/v1");
app.use(upload.any());
eval(req.body.code);

File: test/fixtures/vuln-api.py
from urllib.request import urlopen
from jinja2 import Template
import ldap

url = urlopen(req.body["url"])
t = Template(req.body["template"])
result = ldap.search("cn=" + req.query["name"])
assert req.body["password"] == "secret"
print(f"token: {token}")
User.objects.filter(name=req.GET["name"])
api_url = "http://10.0.0.5:8080/api"
exec(req.body["code"])
